platform/posix/transport/src/mbedtls_pkcs11_posix.c

```diff
@@ -0,0 +1,958 @@
+
+        if( mbedtlsError != 0 )
+        {
+            LogError( ( "Failed to get certificate from PKCS #11 module." ) );
+            returnStatus = MBEDTLS_PKCS11_INVALID_CREDENTIALS;
+        }
+    }
+
+    if( returnStatus == MBEDTLS_PKCS11_SUCCESS )
+    {
+        ( void ) mbedtls_ssl_conf_own_cert( &( pMbedtlsPkcs11Context->config ),
+                                            &( pMbedtlsPkcs11Context->clientCert ),
+                                            &( pMbedtlsPkcs11Context->privKey ) );
+    }
+
+    return returnStatus;
+}
+
+/*-----------------------------------------------------------*/
+
+static MbedtlsPkcs11Status_t configureMbedtlsSniAlpn( MbedtlsPkcs11Context_t * pMbedtlsPkcs11Context,
+                                                      const MbedtlsPkcs11Credentials_t * pMbedtlsPkcs11Credentials,
+                                                      const char * pHostName )
+{
+    MbedtlsPkcs11Status_t returnStatus = MBEDTLS_PKCS11_SUCCESS;
+    int32_t mbedtlsError = 0;
+
+    if( pMbedtlsPkcs11Credentials->pAlpnProtos != NULL )
+    {
+        /* Include an application protocol list in the TLS ClientHello message. */
+        mbedtlsError = mbedtls_ssl_conf_alpn_protocols( &( pMbedtlsPkcs11Context->config ),
+                                                        pMbedtlsPkcs11Credentials->pAlpnProtos );
+
+        if( mbedtlsError != 0 )
+        {
+            LogError( ( "Failed to configure ALPN protocol in MbedTLS: mbedTLSError= %s : %s.",
+                        mbedtlsHighLevelCodeOrDefault( mbedtlsError ),
+                        mbedtlsLowLevelCodeOrDefault( mbedtlsError ) ) );
+            returnStatus = MBEDTLS_PKCS11_INTERNAL_ERROR;
+        }
+    }
+
+    /* Enable SNI if requested. */
+    if( ( returnStatus == MBEDTLS_PKCS11_SUCCESS ) &&
+        ( pMbedtlsPkcs11Credentials->disableSni == false ) )
+    {
+        mbedtlsError = mbedtls_ssl_set_hostname( &( pMbedtlsPkcs11Context->context ),
+                                                 pHostName );
+
+        if( mbedtlsError != 0 )
+        {
+            LogError( ( "Failed to set server name: mbedTLSError= %s : %s.",
+                        mbedtlsHighLevelCodeOrDefault( mbedtlsError ),
+                        mbedtlsLowLevelCodeOrDefault( mbedtlsError ) ) );
+            returnStatus = MBEDTLS_PKCS11_INTERNAL_ERROR;
+        }
+    }
+
+    return returnStatus;
+}
+
+/*-----------------------------------------------------------*/
+
+static MbedtlsPkcs11Status_t configureMbedtlsFragmentLength( MbedtlsPkcs11Context_t * pMbedtlsPkcs11Context )
+{
+    MbedtlsPkcs11Status_t returnStatus = MBEDTLS_PKCS11_SUCCESS;
+
+    /* Set Maximum Fragment Length if enabled. */
+    #ifdef MBEDTLS_SSL_MAX_FRAGMENT_LENGTH
+        int32_t mbedtlsError = 0;
+
+        /* Enable the max fragment extension. 4096 bytes is currently the largest fragment size permitted.
+         * See RFC 8449 https://tools.ietf.org/html/rfc8449 for more information.
```

Comment:
RFC 8449 represents a different extension, the "record size limit" which seems to be the new extension to replace "maximum fragment length". However, this mbed TLS function seems to be for the "maximum fragment length" extension that is defined in [RFC6066](https://datatracker.ietf.org/doc/html/rfc6066#page-8).
```suggestion
        /* Enable the max fragment extension. 4096 bytes is currently the largest fragment size permitted.
         * See RFC 6066 https://tools.ietf.org/html/rfc6066 for more information.
```